### PR TITLE
Add NNCLR pre-trained models

### PR DESCRIPTION
**Pull request by Rishit-dagli**
NNCLR is a self-supervised learning approach as proposed in the paper "With a Little Help from My Friends: Nearest-Neighbor Contrastive Learning of Visual Representations" [1], by Google Research and DeepMind pre-trained on STL-10.

The training code was contributed by me on Keras Examples and could be found [here](https://keras.io/examples/vision/nnclr/).

### References

[1] Dwibedi, Debidatta, et al. “With a Little Help from My Friends: Nearest-Neighbor Contrastive Learning of Visual Representations.” ArXiv:2104.14548 [Cs], Apr. 2021. arXiv.org, http://arxiv.org/abs/2104.14548

---

Any pull request you open is subject to the TensorFlow Hub Terms of Service at www.tfhub.dev/terms and Google's Privacy Policy at https://www.google.com/policies/privacy.

We check modified Markdown files for validity using [this](https://github.com/tensorflow/tfhub.dev/blob/master/.github/workflows/contributions-validator.yml) GitHub Workflow. You can execute the same checks locally by passing the file paths of modified Markdown files (relative to the `assets/docs` directory) to validator.py e.g. `python3 ./tools/validator.py google/google.md google/models/albert_base/1.md ...`.

### CODE REVIEW

File: assets/docs/rishit-dagli/nnclr/1.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+### Overview
+
+NNCLR is a an self supervised learning approach as proposed in the paper "With a Little Help from My Friends: Nearest-Neighbor Contrastive Learning of Visual Representations" [1], by Google Research and DeepMind pre-trained on STL-10.
+
+NNCLR learns self-supervised representations that go beyond single-instance positives, which allows for learning better features that are invariant to different viewpoints, deformations, and even intra-class variations. Clustering based methods offer a great approach to go beyond single instance positives, but assuming the entire cluster to be positives could hurt performance due to early over-generalization. Instead, NNCLR uses nearest neighbors in the learned representation space as positives. In addition, NNCLR increases the performance of existing contrastive learning methods like SimCLR and reduces the reliance of self-supervised methods on data augmentation strategies.
+
+### Usage
+
+This model can be used with the `hub.load` as follows.
```
Comment (WGierke) [inline assets/docs/rishit-dagli/nnclr/1.md:25]: Would you mind adding what dimensions for the input and output are expected?

**Review by WGierke — CHANGES_REQUESTED** [review]
Thanks @Rishit-dagli for providing this model! Would you mind addressing the one comment I left?

File: assets/docs/rishit-dagli/nnclr/1.md (hunk 1 above)
Comment (Rishit-dagli) [reply assets/docs/rishit-dagli/nnclr/1.md:25]: Thanks for spotting this. I originally thought that people could infer the input and output types from the notebook which shows how this model was made but it makes sense to add it in the documentation.

**Comment by Rishit-dagli** [comment]
Thanks @WGierke for the review, added those.

**Review by WGierke — APPROVED** [review]

**Pull request merged**